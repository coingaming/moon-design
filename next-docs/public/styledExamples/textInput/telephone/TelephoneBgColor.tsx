import React from 'react';
import { TextInput } from '@heathmont/moon-core';

const Example = () => (
  <div className="flex flex-col lg:flex-row justify-around items-end w-full gap-2">
    <TextInput
      type="tel"
      inputSize="medium"
      label="medium"
      placeholder="e.g. +372 123 4567"
      hintText="Informative message holder"
      backgroundColor="goku"
    />
    <TextInput
      type="tel"
      inputSize="large"
      label="large"
      placeholder="e.g. +372 123 4567"
      hintText="Informative message holder"
      backgroundColor="goku"
    />
    <TextInput
      type="tel"
      inputSize="xlarge"
      label="xlarge"
      placeholder="e.g. +372 123 4567"
      hintText="Informative message holder"
      backgroundColor="goku"
    />
  </div>
);

export default Example;
