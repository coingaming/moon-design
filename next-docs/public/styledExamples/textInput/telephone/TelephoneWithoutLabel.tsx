import React from 'react';
import { TextInput } from '@heathmont/moon-core';

const Example = () => (
  <div className="flex flex-col lg:flex-row justify-around items-end w-full gap-2">
    <TextInput type="tel" inputSize="medium" placeholder="medium" />
    <TextInput type="tel" inputSize="large" placeholder="large" />
    <TextInput type="tel" inputSize="xlarge" placeholder="xlarge" />
  </div>
);

export default Example;
