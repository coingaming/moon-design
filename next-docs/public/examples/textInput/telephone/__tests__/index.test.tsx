/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { moonDesignLight, ThemeProvider } from '@heathmont/moon-themes';
import TelephoneWithLabel from '../TelephoneWithLabel';
import TelephoneWithoutLabel from '../TelephoneWithoutLabel';
import TelephoneWithHint from '../TelephoneWithHint';
import TelephoneDisabled from '../TelephoneDisabled';
import TelephoneError from '../TelephoneError';
import TelephoneBgColor from '../TelephoneBgColor';
import TelephoneReadonly from '../TelephoneReadonly';

const renderWithTheme = (component: JSX.Element) => (
  <ThemeProvider theme={moonDesignLight}>{component}</ThemeProvider>
);

describe('TextInput: type Tel', () => {
  it('renders with label', () => {
    const testRenderer = renderer.create(
      renderWithTheme(<TelephoneWithLabel />)
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders without label', () => {
    const testRenderer = renderer.create(
      renderWithTheme(<TelephoneWithoutLabel />)
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders with hint', () => {
    const testRenderer = renderer.create(
      renderWithTheme(<TelephoneWithHint />)
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders disabled', () => {
    const testRenderer = renderer.create(
      renderWithTheme(<TelephoneDisabled />)
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders error', () => {
    const testRenderer = renderer.create(renderWithTheme(<TelephoneError />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('supports background colour', () => {
    const testRenderer = renderer.create(renderWithTheme(<TelephoneBgColor />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders readOnly', () => {
    const testRenderer = renderer.create(
      renderWithTheme(<TelephoneReadonly />)
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
