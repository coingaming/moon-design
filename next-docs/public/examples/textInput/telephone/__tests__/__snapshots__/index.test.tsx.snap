// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput: type Tel renders disabled 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 gBLEES"
      disabled={true}
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 bekhms"
        disabled={true}
        placeholder=" "
        type="tel"
        value="+372 123 4567"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 gBLEES"
      disabled={true}
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 czeiwv"
        disabled={true}
        placeholder=" "
        type="tel"
        value="+372 123 4567"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 gBLEES"
      disabled={true}
    >
      <div
        className="Inner-sc-1soq84i-0 QnEHj"
      >
        <input
          className="Input-qkdgvx-0 kwCYox"
          disabled={true}
          placeholder=" "
          type="tel"
          value="+372 123 4567"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Tel renders error 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 GUcka"
        placeholder="e.g. +372 123 4567"
        type="tel"
      />
      <p
        className="HintText-sc-1e9hzxr-0 heGnbI"
        role="alert"
      >
        Error message
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 gYHLjZ"
        placeholder="e.g. +372 123 4567"
        type="tel"
      />
      <p
        className="HintText-sc-1e9hzxr-0 heGnbI"
        role="alert"
      >
        Error message
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 QnEHj"
      >
        <input
          className="Input-qkdgvx-0 ekIlEn"
          placeholder="e.g. +372 123 4567"
          type="tel"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 heGnbI"
        role="alert"
      >
        Error message
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Tel renders readOnly 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 bekhms"
        placeholder=" "
        readOnly={true}
        type="tel"
        value="+372 123 4567"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 czeiwv"
        placeholder=" "
        readOnly={true}
        type="tel"
        value="+372 123 4567"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 QnEHj"
      >
        <input
          className="Input-qkdgvx-0 kwCYox"
          placeholder=" "
          readOnly={true}
          type="tel"
          value="+372 123 4567"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Tel renders with hint 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 bekhms"
        placeholder="e.g. +372 123 4567"
        type="tel"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 czeiwv"
        placeholder="e.g. +372 123 4567"
        type="tel"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 QnEHj"
      >
        <input
          className="Input-qkdgvx-0 kwCYox"
          placeholder="e.g. +372 123 4567"
          type="tel"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Tel renders with label 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 bekhms"
        placeholder="e.g. +372 123 4567"
        type="tel"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 czeiwv"
        placeholder="e.g. +372 123 4567"
        type="tel"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 QnEHj"
      >
        <input
          className="Input-qkdgvx-0 kwCYox"
          placeholder="e.g. +372 123 4567"
          type="tel"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Tel renders without label 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <input
        className="Input-qkdgvx-0 bekhms"
        placeholder="medium"
        type="tel"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <input
        className="Input-qkdgvx-0 czeiwv"
        placeholder="large"
        type="tel"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 QnEHj"
      >
        <input
          className="Input-qkdgvx-0 eBLNew"
          placeholder="xlarge"
          type="tel"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Tel supports background colour 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 jsDDWX"
        placeholder="e.g. +372 123 4567"
        type="tel"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 rTkGE"
        placeholder="e.g. +372 123 4567"
        type="tel"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 hrjkOU"
      >
        <input
          className="Input-qkdgvx-0 kwCYox"
          placeholder="e.g. +372 123 4567"
          type="tel"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;
